include/osquery/dispatcher.h

```diff
@@ -115,7 +115,7 @@ class InternalRunnable : private boost::noncopyable,
 
 /// An internal runnable used throughout osquery as dispatcher services.
 using InternalRunnableRef = std::shared_ptr<InternalRunnable>;
-using InternalThreadRef = std::shared_ptr<std::thread>;
+using InternalThreadRef = std::unique_ptr<std::thread>;
```

Comment:
Why are you doing this?

Reply:
It was believed that changing shared_ptr to unique_ptr caused the memory leak so I am reverting this change(https://github.com/facebook/osquery/pull/4597) and proving that this change is not the reason of memory leak, besides unique_ptr is more relevant in this context than shared_ptr.